spring-kafka/src/test/java/org/springframework/kafka/annotation/EnableKafkaIntegrationTests.java

```diff
@@ -976,6 +1017,36 @@ public KafkaListenerErrorHandler voidSendToErrorHandler() {
 			};
 		}
 
+		private Throwable listen16Exception;
+
+		private Object listen16Message;
+
+		private CountDownLatch listen16ErrorLatch = new CountDownLatch(1);
+
+		@Bean
+		public ConsumerAwareErrorHandler listen16ErrorHandler() {
+			return (e, r, c) -> {
+				listen16Exception = e;
+				listen16Message = r.value();
+				listen16ErrorLatch.countDown();
+			};
+		}
+
+		@Bean
+		public FooConverter fooConverter() {
+			return new FooConverter();
+		}
+
+		@Bean
+		public ConversionService conversionService() {
+			ConversionServiceFactoryBean bean = new ConversionServiceFactoryBean();
+			bean.setConverters(Collections.singleton(
+					fooConverter()
+			));
+			bean.afterPropertiesSet();
+			return bean.getObject();
```

Comment:
You don't need to call `afterPropertiesSet()` in the `@Bean` definition method.
You don't need to call `FactoryBean.getObject()` in the `@Bean` definition method.
The Framework takes care about all the lifecycle callbacks and autowiring injections.